### PR TITLE
fix #61, fix #60 map and arrow rotation

### LINKED ISSUE #60: Compass as marker?
I really love the compass that we have!
However, the current implementation is not really useful for the PiP mode (looks kind of not really cool and that's why it is hidden; also due to the lack of space).

I have no clue how mapsforge works, but could not we mark the current position (the one that is currently marked with a red dot) similar as we put the marker(s) on the map?
I mean the marker pins are zoom invariant and do not seem to incur rendering performance issues, right?
Long story short: can render the compass arrow as a "marker" and just rotate it?
And when we get a new position, we either move or delete/add the compass arrow again.

PS: I was just thinking aloud.

### LINKED ISSUE #61: Auto-rotate map based on your compass heading.
Do you find useful this option? So users could select between:
- North-up.
- Track-up.

### PR DESCRIPTION
**Pull request by pstorch**
Before merging and closing the issues again, I would like to have some tests from @dennisguse and @rgmf 

The Map and the Arrow have 3 different modes (NORTH, COMPASS and DIRECTION), both can be changed via the Menu and combined with each other.

Very interesting for me will be if the map and the arrow rotates live at @dennisguse 's device.

### CODE REVIEW

**Comment by rgmf** [comment]
Installed and ready to test... I'll give you my feedback ;)

**Comment by dennisguse** [comment]
Works for me (except that my compass sucks, but that's not related...).
TL;DR Love it.*

Thoughts:
1. we need some non-linear smoothing especially if the compass is used (arrow + map).
Especially with the compass it is pretty hard.
Trail sense has some nice smoothing: https://github.com/kylecorry31/Trail-Sense
2. why do we need north up for the arrow?
Is this useful? I did not get how it helps me navigating - I guess I am just not used to this functionality.
As a default option, I would say let's use dot (e.g., directionless) instead of arrow north.

PS: I now understood your discussion about traveling direction vs. compass orientation.
I am not sure that we need to configure them both separately (for map vs. arrow), but I really like this as my compass is literally always completely off (actually on both phones ><).

**Comment by pstorch** [comment]
Thanks for the feedback. 
For 1) I've opened a separate issue. 
Regarding 2): e.g. if you rotate the map into moving direction, but want to know where north is

**Comment by dennisguse** [comment]
@rgmf Wow - I did not see this one ><

**Comment by pstorch** [comment]
@rgmf ha ha, that is a really good one :rofl:

**Comment by dennisguse** [comment]
@pstorch And you picked exactly the correct smiley ><
